provider/aws/aws.go

```diff
@@ -324,69 +323,66 @@ func (p *AWSProvider) zones(ctx context.Context) (map[string]*profiledZone, erro
 	log.Debug("Refreshing zones list cache")
 
 	zones := make(map[string]*profiledZone)
-	var profile string
-	var tagErr error
-	f := func(resp *route53.ListHostedZonesOutput, lastPage bool) (shouldContinue bool) {
-		for _, zone := range resp.HostedZones {
-			if !p.zoneIDFilter.Match(aws.StringValue(zone.Id)) {
-				continue
-			}
 
-			if !p.zoneTypeFilter.Match(zone) {
-				continue
-			}
+	for profile, client := range p.clients {
```

Comment:
I think we should try to reduce the change. This seems a bit too much unrelated for the aws sdk version and please try to only change the parts we really need to change.

Reply:
This isn't an intentional change and I followed the existing logic as much as possible, it's due to `ListHostedZonesPagesWithContext` not having a direct peer in awsk-sdk-go-v2 https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/route53

In this case, the refactor was to move to `ListHostedZones` in aws-sdk-go-v2 and use its paginator https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#using-paginators

Reply:
ok